### DISCUSSION TITLE
Copy / Paste ALL items

### DISCUSSION THREAD

**Discussion by AndremBonetti** [musescore/MuseScore#13607] [Ideas]
When composing or scoring I often move entire sections. EG: I'll write a "DS al Coda", and later decide to copy what _was_ the segno to later in the piece so I can re-score it. 
Presently, when you copy / paste a section it copies the dots, but not all the other information such as tempo markings; repeat signs; and time signatures. Instead, the notes jumble into the wrong time signature / key signature. 

Is there any scope to make this possible? I know this has been suggested a few times over the years. I've pasted a few links below. 

Much appreciations!

![Musescore_CopyPaste](https://user-images.githubusercontent.com/114966370/193718524-1dcea2e5-b05f-4c02-a55b-9aeaffc903a5.jpg)

https://musescore.org/en/node/332502
https://www.reddit.com/r/Musescore/comments/mhhnye/is_there_any_way_to_copy_and_paste_all_aspects_of/

**Comment by AndremBonetti** [comment]
PS: I'm still using Sibelius 5 (Sib7 is shocking, Dorico is a good contender but if it's anything like Cubase they'll keep releasing new versions which should be a patch). 

I usually do stupid stuff such as spelling morse code in time signatures, then copy it a few times throughout the score. So I just can't practically use Musescore without a more inclusive copy/paste function. 

![image](https://user-images.githubusercontent.com/114966370/193720793-b0e5a21a-aa70-46ed-8048-888e46bd5558.png)

**Comment by Kqpout** [comment]
Yes! Those repeat signs not copypasting are also a great source of frustration for me.

**Comment by AndremBonetti** [comment]
Hi all,

Stoked that Musescore 4 is out!
I haven't had time to familiarise myself with it yet, but suspect if my suggestion above was taken on board it will take me some time to find it. 

Does anyone know if this has been addressed?

Cheers,
Andre

**Comment by TeeDeeY** [comment]
I made a time signature copy and paste/overwrite plug-in for Musescore.  It has some limitations.  The limitations are listed on Musescore.org.  The download link goes to my GitHub.  I just finished the alpha version and will update as I learn how to do more things.  

I have a preliminary handle on how to copy tempo and rehearsal marks.  There will be a copy and paste for them in a maybe a couple months.  I have no idea on how “repeats” are stored and accessed by the plug-in but I can at least look into the issue in a couple of months.  

I have to keep investigating the plug-in environment to see how things can be done.  Some are possible for me and some are not. 

https://musescore.org/en/project/time-signature-copy-paste

Please download it and try it.  Make sure to have sufficient measures for it to overwrite, as mentioned in the description. 

I will do what I can and others reading the plugins could make further inroads and make their own improved version.
